provisioning/device/src/polling_state_machine.ts

```diff
@@ -6,17 +6,17 @@
 import { EventEmitter } from 'events';
 import { errors } from 'azure-iot-common';
 import * as machina from 'machina';
-import * as Provisioning from './interfaces';
+import { ProvisioningAuthentication, PollingTransportHandlers } from './interfaces';
 import * as dbg from 'debug';
 const debug = dbg('azure-device-provisioning:transport-fsm');
 
-export class  ClientStateMachine extends EventEmitter {
+export class  PollingStateMachine extends EventEmitter {
```

Comment:
>PollingStateMachine [](start = 14, length = 19)

the authorization object we carry around make this very x509 specific. is there a way to make this polling state machine agnostic from x509 or TPM? we are going to have the same need with TPM except the authorization will be a SAS token.

Reply:
It should be agnostic already AFAIK.  This is what we're passing around (taken from restApiclient): 
export type ProvisioningAuthentication = string | X509 | SharedAccessSignature;

---
In reply to: [151320986](https://github.com/Azure/azure-iot-sdk-node/pull/154#discussion_r151320986) [](ancestors = 151320986)